apps/champions/lib/champions/users.ex

```diff
@@ -195,4 +206,37 @@ defmodule Champions.Users do
       {:error, _, reason, _} -> {:error, reason}
     end
   end
+
+  @doc """
+  Level up the Kaline Tree of a user.
+  """
+  def level_up_kaline_tree(user_id) do
+    with {:user, {:ok, user}} <- {:user, Users.get_user(user_id)},
+         level_up_costs = calculate_costs_to_level_up_kaline_tree(user),
+         {:can_afford, true} <-
+           {:can_afford,
+            Enum.all?(level_up_costs, fn %CurrencyCost{currency_id: level_up_currency_id, amount: level_up_cost} ->
+              Currencies.can_afford(
+                user_id,
+                level_up_currency_id,
+                level_up_cost
+              )
+            end)} do
+      Users.level_up_kaline_tree(user_id, level_up_costs)
+    else
+      {:can_afford, false} -> {:error, :cant_afford}
+    end
+  end
+
+  defp calculate_costs_to_level_up_kaline_tree(user),
```

Comment:
Let's leave a comment here linking to a GH issue explaining the context for when we replace this!